src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/SearchQueryFactoryOperation.java

```diff
@@ -60,14 +68,24 @@ public Query build(MetaData metadata, Map<String, String> params, boolean isCoun
         return query;
     }
 
-    private String secure(String clauses) {
+    private String secure(String clauses, MetaData metadata) {
+        String newClauses = clauses;
         if (UserAuthorisation.AccessLevel.GRANTED.equals(userAuthorisation.getAccessLevel())) {
-            clauses += String.format(
+            newClauses += String.format(
                 " AND id IN (SELECT cu.case_data_id FROM case_users AS cu WHERE user_id = '%s')",
                 userAuthorisation.getUserId()
             );
         }
-        return clauses;
+
+        // restrict cases to the case states the user has access to
+        List<CaseState> caseStates = authorisedCaseDefinitionDataService.getUserAuthorisedCaseStates(metadata.getJurisdiction(),
+                                                                                                     metadata.getCaseTypeId(),
+                                                                                                     CAN_READ);
+        if (CollectionUtils.isNotEmpty(caseStates)) {
+            newClauses += String.format(" AND state IN ('%s')", String.join("','", caseStates.stream().map(CaseState::getId).collect(Collectors.toList())));
```

Comment:
shall we also add a method to AuthorisedCaseDefinitionDataService to retrieve only the state ids rather than having to 
```caseStates.stream().map(CaseState::getId).collect(Collectors.toList()))```
We are doing this in 2 places already, so maybe it's worth considering

Reply:
good point. Done that